feat(api-reference): version prefix

hanspagel opened the pull request:
**Problem**
Currently, a lot of people just having something like `1` or `2` as the version in their API definition, but I think the rendering might be a bit confusing. The number is just shown without any context.

<img width="364" alt="Screenshot 2025-01-14 at 12 07 53" src="https://github.com/user-attachments/assets/d2fa844a-a4bd-4e7d-9a44-abc7bacdb89f" />


**Solution**
With this PR it’s rendered as `v1`, which might be enough to make it look like a version number.

<img width="364" alt="Screenshot 2025-01-14 at 12 07 39" src="https://github.com/user-attachments/assets/e16690ae-d9a1-49b0-a490-363b0d40961f" />

Wdyt?

**Checklist**

I’ve gone through the following:

- [x] I’ve added an explanation _why_ this change is needed.
- [x] I’ve added a changeset (`pnpm changeset`).
- [x] I’ve added tests for the regression or new feature.
- [ ] I’ve updated the documentation.

antlio reviewed (approved):
great addition

amritk commented on packages/api-reference/src/components/Content/Introduction/Introduction.vue:
```diff
@@ -38,10 +38,8 @@ const formattedSpecTitle = computed(() => {
-            {{ info.version }}
-          </Badge>
-          <Badge v-if="specVersion"> OAS {{ specVersion }}</Badge>
+          <Badge v-if="info.version">v{{ info.version }}</Badge>
```
```suggestion
          <Badge v-if="info.version">v{{ info.version.replace(/^v/, '') }}</Badge>
```
should we do a quick check in case they already have a v? to avoid a `v v 1`

hanspagel replied:
Man, that’s smart!

Let us take a different approach: We should check if the first string is a number, otherwise we could end up with `vstaging` and other funny things.

Also, currently it’s possible a non-string value lands there, so we need to make sure it’s a string.

I’ll add this tomorrow!

amritk reviewed (approved):
approved with minor change

hanspagel commented:
Updated the component:

* to not prefix the version when the first character isn't a number (e.g. `beta`, `staging` or whatever),
* not show a version if it’s not a string, but whatever other data type
* add tests for the whole Introduction component
* remove unused CSS
* replaced one CSS class with tailwind (gap: 48px > gap-12)
* renamed some variables
* added some components

antlio commented:
should we add test and support on `v1.0` usage for instance?

amritk commented on packages/api-reference/src/components/Content/Introduction/Introduction.vue:
```diff
@@ -23,33 +23,48 @@ const props = defineProps<{
+/**
+ * Get the OpenAPI/Swagger specification version from the API definition.
+ */
+const oasVersion = computed(() => {
+  return props.parsedSpec?.openapi ?? props.parsedSpec?.swagger ?? ''
 })
```
```suggestion
const oasVersion = computed(() => props.parsedSpec?.openapi ?? props.parsedSpec?.swagger ?? '')
```
the beauty of arrow functions is no need to write return!

amritk commented on packages/api-reference/src/components/Content/Introduction/Introduction.vue:
```diff
@@ -23,33 +23,48 @@ const props = defineProps<{
+const version = computed(() => {
+  // Prefix the version with “v” if the first character is a number, don’t prefix if it’s not.
+  // Don’t output anything when version is not a string.
+  return typeof props.info?.version === 'string'
+    ? props.info.version.toString().match(/^\d/)
+      ? `v${props.info.version}`
+      : props.info.version
+    : undefined
```
```suggestion
props.info?.version?.toString().startsWith('0-9') ? `v${props.info.version}` : props.info?.version
```
this version will handle numbers instead of returning undefined. completely up to you though!

hanspagel replied:
`startsWith(0-9)` checks for the literal `0-9`, not for numbers between 0 and 9 😬 

and without the string check it would output arrays, objects, booleans … (which no one should pass, but doesn’t hurt to add the check)

amritk commented on packages/api-reference/src/features/DownloadLink/DownloadLink.vue:
```diff
@@ -11,8 +11,14 @@ const props = defineProps<{
 
-const getHideDownloadButtonSymbol = inject(HIDE_DOWNLOAD_BUTTON_SYMBOL)
-const getOpenApiDocumentUrlSymbol = inject(OPENAPI_DOCUMENT_URL_SYMBOL)
+const getHideDownloadButtonSymbol = inject(
```
I removed these in my other pr

amritk reviewed (approved):
lol im commenting on this pr then i see its already merged!
